apis/management.cattle.io/v3/authn_types.go

```diff
@@ -146,8 +146,44 @@ type ActiveDirectoryConfig struct {
 }
 
 type ActiveDirectoryTestAndApplyInput struct {
-	ActiveDirectoryConfig ActiveDirectoryConfig `json:"activeDirectoryConfig, omitempty"`
+	ActiveDirectoryConfig ActiveDirectoryConfig `json:"activeDirectoryConfig,omitempty"`
 	Username              string                `json:"username"`
 	Password              string                `json:"password"`
 	Enabled               bool                  `json:"enabled,omitempty"`
 }
+
+type OpenLDAPConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	AuthConfig        `json:",inline" mapstructure:",squash"`
+
+	Servers                     []string `json:"servers,omitempty"                     norman:"type=array[string],required"`
+	Port                        int64    `json:"port,omitempty"                        norman:"default=389,required"`
+	TLS                         bool     `json:"tls,omitempty"                         norman:"default=false,required"`
+	Certificate                 string   `json:"certificate,omitempty"`
+	ServiceAccountUsername      string   `json:"serviceAccountUsername,omitempty"      norman:"required"`
+	ServiceAccountPassword      string   `json:"serviceAccountPassword,omitempty"      norman:"required"`
+	UserDisabledBitMask         int64    `json:"userDisabledBitMask,omitempty"         norman:"default=2"`
+	UserSearchBase              string   `json:"userSearchBase,omitempty"              norman:"required"`
+	UserSearchAttribute         string   `json:"userSearchAttribute,omitempty"         norman:"default=uid,required"`
```

Comment:
We did an enhancement to search for AD in 2.0. I'd like you to follow the same pattern for LDAP. Basically, more than one searchable attribute can be defined using `|` as a delimeter:
```
 	UserSearchAttribute         string   `json:"userSearchAttribute,omitempty"         norman:"default=sAMAccountName|sn|givenName,required"`
```
This is a low priority change, so if there are other things to change as part of the review, you can change those first.